packages/workspace/src/generators/convert-to-buildable/convert-to-buildable.ts

```diff
@@ -0,0 +1,175 @@
+    ...(schema.type === 'angular'
+      ? {
+          peerDependencies: {
+            '@angular/common': packageJson['dependencies']['@angular/common'],
+            '@angular/core': packageJson['dependencies']['@angular/core'],
+          },
+          dependencies: { tslib: packageJson['dependencies']['tslib'] },
+        }
+      : {}),
+  });
+
+  // Write the build target to the projects configuration
+  const buildTarget: any = {
+    executor: getExecutor(schema.type),
+    outputs: ['{options.outputPath}'],
+    options: {
+      outputPath: `dist/${configuration.root}`,
+      tsConfig: `${configuration.root}/tsconfig.lib.json`,
+      main: `${configuration.root}/src/index.ts`,
+      assets: [`${configuration.root}/*.md`],
+    },
+  };
+
+  switch (schema.type) {
+    case 'nest':
+    case 'node':
+      buildTarget.options['packageJson'] = `${configuration.root}/package.json`;
+      break;
+    case 'next':
+    case 'react':
+      const { main, ...options } = buildTarget.options;
+      buildTarget.options = {
+        ...options,
+        project: `${configuration.root}/package.json`,
+        entryFile: `${configuration.root}/src/index.ts`,
+        external: ['react/jsx-runtime'],
+        rollupConfig: '@nrwl/react/plugins/bundle-rollup',
+        compiler: 'babel',
+        assets: [
+          {
+            glob: `${configuration.root}/README.md`,
+            input: '.',
+            output: '.',
+          },
+        ],
+      };
+      break;
+    case 'angular':
+      buildTarget.options = {
+        project: `${configuration.root}/ng-package.json`,
+      };
+      buildTarget.outputs = [`dist/${configuration.root}`];
+      buildTarget.configurations = {
+        production: {
+          tsConfig: `${configuration.root}/tsconfig.lib.prod.json`,
+        },
+        development: {
+          tsConfig: `${configuration.root}/tsconfig.lib.json`,
+        },
+      };
+      buildTarget.defaultConfiguration = 'production';
+      break;
+  }
+
+  if (schema.type === 'angular') {
+    writeJson(host, join(configuration.root, 'tsconfig.lib.prod.json'), {
+      extends: './tsconfig.lib.json',
+      compilerOptions: {
+        declarationMap: false,
+      },
+      angularCompilerOptions: {},
+    });
+
+    writeJson(host, join(configuration.root, 'ng-package.json'), {
+      $schema: '../../node_modules/ng-packagr/ng-package.schema.json',
+      dest: `../../dist/${configuration.root}`,
+      lib: {
+        entryFile: 'src/index.ts',
+      },
+    });
+  }
+
+  const isStandalone = isStandaloneProject(host, schema.project);
+
+  if (isStandalone) {
+    writeJson(host, join(configuration.root, 'project.json'), {
+      ...configuration,
+      targets: {
+        ...configuration.targets,
+        build: buildTarget,
+      },
+    });
+  } else {
+    updateJson(host, getWorkspacePath(host), (value) => {
+      value.projects[schema.project].targets['build'] = buildTarget;
+      return value;
+    });
+  }
```

Comment:
You should be able to do this with the `updateProjectConfiguration` helper function, then you don't need to worry if this is standalone or not as the helper function will take care of that for you.